tests/integ/test_reduce_describe_query.py

```diff
@@ -0,0 +1,46 @@
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+import pytest
+
+from snowflake.snowpark.functions import col
+
+pytestmark = [
+    pytest.mark.skipif(
+        "config.getoption('local_testing_mode', default=False)",
+        reason="Reducing describe queries is not supported in Local Testing",
+    ),
+]
+
+
+paramList = [True, False]
+
+
+@pytest.fixture(params=paramList, autouse=True)
+def setup(request, session):
+    is_reduce_describe_query_enabled = session.reduce_describe_query_enabled
+    session.reduce_describe_query_enabled = request.param
+    yield
+    session.reduce_describe_query_enabled = is_reduce_describe_query_enabled
+
+
+@pytest.mark.parametrize(
+    "action",
+    [
+        lambda df: df.filter(col("a") > 2),
+        lambda df: df.sort(col("a").desc()),
+        lambda df: df.limit(2),
+        lambda df: df.filter(col("a") > 2).sort(col("a").desc()).limit(2),
+        lambda df: df.sample(0.5),
+        lambda df: df.sample(0.5).filter(col("a") > 2),
+        lambda df: df.filter(col("a") > 2).sample(0.5),
+    ],
+)
+def test_schema_no_change(session, action):
```

Comment:
Is there a way to count the number of describe queries so you can assert they are 0 (or whatever the reduced number is) as part of the test?